sdk/eventhub/azure-eventhubs/azure/eventhub/common.py

```diff
@@ -57,7 +57,7 @@ class EventData(object):
     PROP_TIMESTAMP = b"x-opt-enqueued-time"
     PROP_DEVICE_ID = b"iothub-connection-device-id"
 
-    def __init__(self, body=None, **kwargs):
+    def __init__(self, body=None, to_device=None, message=None):
```

Comment:
This code is very confusing - if I specify both a `body` and a `message`, the `body` will be ignored. And silently ignored at that. 

If we need to distinguish between the two, we could have a class method `EventData.from_message(message)`. But the way it is implemented now is problematic.

Reply:
This **message** parameter is actually for constructing eventdata when amqp messages are received from service side.
Users shouldn't use it directly (actually user shouldn't be reaching any stuff related to the amqp library).
I've added a new private method **_from_message** in EventData used for internally constructing EventData from received message.